src/ER_V/erv_arm/erv.cpp

```diff
@@ -23,32 +26,89 @@ Scorbot::~Scorbot()
     if(close(fd)) LOG_ERROR("Scorbot: Could not close device descriptor: %s", strerror(errno));
 }
 
+/**
+ * @brief Helper function that determines if a character is a line terminator for the Scorbot RX line
+ * @param ch Character to determine
+ * @returns 1 if a terminating char, 0 otherwise.
+*/
+static uint8_t is_term (char ch)
+{
+    switch (ch)
+    {
+    // Intentional fallthroughs
+        case '\r':
+        case '\n':
+        case '\0':
+        case '>':
+            return 1;
+        default:
+            break;
+    }
+
+    return 0;
+}
+
+/**
+ * @brief Flushes the receive buffer, starting from index 0, up to index length; Logs the flushed data
+ * @details Assumes buffer length of ERV_TTY_BUFFER_LEN
+ * @param tty_buffer The buffer to flush; intended to be the static buffer storing the rx data
+ * @param len Number of chars to flush
+*/
+static void flush_buffer(char* tty_buffer, uint16_t len)
+{
+    if (!tty_buffer) STD_FAIL_VOID;
+    if (!len) return;
+
+    tty_buffer[ERV_TTY_BUFFER_LEN - 1] = '\0';      // Ensure end is null terminated
+    if (len + 1 < ERV_TTY_BUFFER_LEN) tty_buffer[len + 1] = '\0';
+
+    LOG_VERBOSE(0, "SCOR: %s", tty_buffer);
+    memset(tty_buffer, 0, len);
+}
+
+/**
+ * @brief Function that is called by the parent Arm thread in a timed loop
+ * @details Primary purpose is to log data received from Scorbot.
+ * Flushes buffer after receiving a line terminator, or after a fixed length of time
+*/
 void Scorbot::Poll()
 {
-    const uint8_t size = 255;
-    static uint8_t len = 0;
-    uint8_t start = len;
-    static char buffer[size];
-    // LOG_INFO("LEN: %d", len);
-    memset(&buffer[0], 0xFF, size);
-    len += read(fd, &buffer[0], size);
-    // LOG_INFO("LEN: %d", len);
-
-    for (uint8_t iter = start; iter < len && iter < size; iter++)
+    static clock_t last_print;
+    static char buffer[ERV_TTY_BUFFER_LEN];
+    static uint16_t len = 0;
+
+    // Handle new info in buffer
+    char inbox[ERV_TTY_BUFFER_LEN];
+    int result = read(fd, &inbox[0], ERV_TTY_BUFFER_LEN);
+    if (-1 != result)
     {
-        if ('\n' == buffer[iter] || '\0' == buffer[iter])
+        for (uint16_t iter = 0; iter < result; iter++)
         {
-            buffer[iter] = '\0';
-            LOG_INFO("SCOR: %s", buffer);
-            strcpy(&buffer[0], &buffer[start]);
-            len = iter - start;
+            if (is_term(inbox[iter]))
+            {
+                flush_buffer(buffer, len);
+                len = 0;
+                last_print = clock();
+                continue;
+            }
+
+            buffer[len++] = inbox[iter];
         }
     }
+
+    // Handle rx timeout; If timeout has occurred, flush buffer
+    double delta_time_ms = ((double)(clock() - last_print) * 1000) / (CLOCKS_PER_SEC);
+    if (len > 0 && (float) ERV_RX_TIMEOUT_MS < delta_time_ms)
```

Comment:
If we're in cpp, lets make a habit to use static_cast and the like instead of C style.